### DISCUSSION TITLE
Git for Windows v2.47.0.windows.2

### DISCUSSION THREAD

**Discussion by gitforwindowshelper** [git-for-windows/git#5224] [Announcements]
Changes since Git for Windows v2.47.0 (October 8th 2024)

### Bug Fixes

* A regression in v2.47.0 where `git maintenance start` crashed immediately [was fixed](https://github.com/git-for-windows/git/pull/5198).
* A [regression](https://github.com/git-for-windows/git/issues/5199) where clones, fetches and pushes via SSH would dead-lock was fixed.
* As of Git for Windows v2.47.0, [Scalar](https://git-scm.com/docs/scalar) was supposed to enable [an optimized push algorithm](https://github.com/git-for-windows/git/pull/5171), but for a typo didn't, which [was fixed](https://github.com/git-for-windows/git/pull/5220).
* A few documentation and other, minor bug fixes from upstream Git [were integrated into Git for Windows early](https://github.com/git-for-windows/git/pull/5221).

Filename | SHA-256
-------- | -------
Git-2.47.0.2-64-bit.exe | 83c618dcc50b63f0f7afc86a7a125169bd59559ff680683f1d915c45e05ff4cc
Git-2.47.0.2-32-bit.exe | b798dc1ff11afe912994e0e7d0b538edbbd291e666fdc6f34fa4cb19547aefc7
PortableGit-2.47.0.2-64-bit.7z.exe | c77368a8f6ccbd43bde0df0ab603133ce885407a018787d6f1971e040590f1ab
PortableGit-2.47.0.2-32-bit.7z.exe | 71d7de445ce2412c0344f5258dec4a51cac3b4b536b0d68c8e753729d429d365
MinGit-2.47.0.2-64-bit.zip | c4a5d3a2adda98b25fe59349733fca56b3843360b962dba5535282a9d8120b31
MinGit-2.47.0.2-32-bit.zip | 59748b3193b6f27849e35ee76165d53f47a5a205a4cd83e926800570aeaf5aa2
MinGit-2.47.0.2-busybox-64-bit.zip | 672aa54dd79d35baa75df934e2bb6faad5b9f45a99339b1dbd5a1fb2f8b44348
MinGit-2.47.0.2-busybox-32-bit.zip | 94ff74c2c7f68a5093b57c527398fe176230adadb71c25722f0c43a5de611f7b
Git-2.47.0.2-64-bit.tar.bz2 | b6ca6dcd5c818396faa57e06e10489aed3e16396317475ca8e88e30e4eb2e3c5
Git-2.47.0.2-32-bit.tar.bz2 | d861cdef6ef1c93f412eb6e5fa766973a6eee5f8a08973ac4a396bb235c3d489

<hr /><em>This discussion was created from the release <a href='https://github.com/git-for-windows/git/releases/tag/v2.47.0.windows.2'>Git for Windows v2.47.0.windows.2</a>.</em>

**Comment by kevincharp** [comment]
I would like to express my dissatisfaction with a recent issue I encountered while attempting to clone multiple repositories from github and gitlab using SSH. Despite some repositories cloning successfully, others consistently stalled during the process. After thorough investigation, I discovered that this problem stemmed from the latest update, specifically Git for Windows v2.47.0.windows.1.

What particularly frustrates me is that this version is not available through winget, forcing users like myself to manually download the installer from the repository. This not only creates an unnecessary hurdle for individuals but also for teams and organizations relying on seamless updates for their development workflows.

I urge you to prioritize the availability of the latest versions through winget to ensure that all users, including those in my organization, can update efficiently without encountering such issues in the future.

![image](https://github.com/user-attachments/assets/eb9d1e1c-38d9-4e2b-ae15-847714d5319c)

![image](https://github.com/user-attachments/assets/574df2fc-fc33-4f3b-839f-9f611dd4dbc8)

**Reply by rimrul** [reply]
> I urge you to prioritize the availability of the latest versions through winget

No. WinGet package definitions are provided by a third party not associated with the Git for Windows project. Due to their process it takes them about a week to release a new package  once we release an installer, but that's not something we can change.

If you want swift updates, the installer is always the fastest way and the only supported way. If you insist on using a package manager that redistributes our releases in some form, chocolatey and scoop usually both have a package available within a few hours after a release.

**Reply by kevincharp** [reply]
Thank you for your response. I was completely unaware that the update of the package via winget depends on a third party. It's unfortunate how slowly they act to release the most updated version, as it can lead to many conflicts for users during that time.

Who should this type of complaint or suggestion be addressed to, specifically?

**Reply by dscho** [reply]
A better idea than complaining in a way that risks as coming over as entitled would be to become part of the solution rather than of the problem.

In the case of git-scm.com, it seems that the automation is not picking up the new version, which is a bug. It _may_ be that the version comparison at https://github.com/git/git-scm.com/blob/6b38d7cb251489217d9a9089f3c8521cd49cf9fa/script/update-download-data.rb#L50 is buggy.

**Reply by rimrul** [reply]
> In the case of git-scm.com, it seems that the automation is not picking up the new version

That does say "source release" and refers to a git.git version

**Reply by kevincharp** [reply]
@dscho My intention has always been to raise a suggestion regarding a problem I encountered and to notify others about it. I would love to contribute technically and share my knowledge towards a solution for this issue. However, the only way I see to do this at the moment is by informing the people involved in the projects.

I apologize if my message came across as just a simple complaint. I just learned something new about how long the release process takes to be approved by package managers like Winget.

You might be in a position where your technical skills and knowledge surpass mine, but I always come with the intention to learn.

My comment was simply a frustration I encountered between yesterday and today.

**Comment by lakshminarayana02** [comment]
Automation is not taking up the updated version, which is affecting  huge number of clients

https://github.com/actions/runner-images/blob/23478d3106e1ce292473313bc885a6e861caaaa0/images/windows/scripts/build/Install-Git.ps1#L9

However, the version still shows as Git 2.47.0.windows.1
https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md

**Reply by dscho** [reply]
@lakshminarayana02 I suspect you wanted to report this to `actions/runner-images`. Git for Windows has no control whatsoever over the hosted runner images of GitHub Actions (I sometimes wish we had at least _some_ influence, e.g. to have access to Windows/ARM64 runners, but we don't have any influence at all as to what they are doing over there).

**Comment by Oranyelu** [comment]
I have tried countlessly to download the git .exe file from these assets links and the link on the main website but I keep encountering an error.
I have tried this on multiple devices and still have the same error.

**Reply by Oranyelu** [reply]
here i s the error page i get 
![Uploading image.png…]()

**Reply by Oranyelu** [reply]
![Screenshot 2024-11-24 204837](https://github.com/user-attachments/assets/bf8cbbf0-5783-4dc0-8165-7ac18e89f608)

**Reply by dscho** [reply]
"Timed out" sounds like a problem with GitHub or your network connection. I don't see anything obvious in https://www.githubstatus.com/history though.
